src/main/java/frc/robot/subsystems/drive/DriveSubsystem.java

```diff
@@ -309,9 +289,11 @@ public void keepHeading(double x, double y, boolean fieldRelative) {
     double headingDifferenceDegrees = currentHeadingDegrees - targetHeadingDegrees;
     double offsetHeadingDegrees = MathUtil.inputModulus(headingDifferenceDegrees, -180, 180);
 
-    double desiredRotation =
-        Cal.SwerveSubsystem.ROTATE_TO_TARGET_PID_CONTROLLER.calculate(offsetHeadingDegrees, 0.0)
-            + Math.signum(offsetHeadingDegrees) * Cal.SwerveSubsystem.ROTATE_TO_TARGET_FF;
+    double pidRotation =
+        Cal.SwerveSubsystem.ROTATE_TO_TARGET_PID_CONTROLLER.calculate(offsetHeadingDegrees, 0.0);
+    double ffRotation = Math.signum(offsetHeadingDegrees) * Cal.SwerveSubsystem.ROTATE_TO_TARGET_FF;
+
+    double desiredRotation = pidRotation - ffRotation;
```

Comment:
I can't remember, and I'm on my phone so looking deeper at the code is hard. But confirm that subtracting is the correct operation, since in the old code we are adding.

Reply:
Yep this changed intentionally, and tested at the practice field saturday, based on the bug from when we walked through everything at the event